shencha-master/shencha-master/agent/extract_agent.py

```
@@ -1,87 +1,231 @@
+            if not response or 'choices' not in response:
+                raise ValueError("Invalid API response format")
+                
+            content = response['choices'][0]['message']['content']
+            if content.startswith("```json"):
+                content = content.strip("```json").strip("```")
+            elif content.startswith("```"):
+                content = content.strip("```")
+                
+            result = json.loads(content.strip())
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Failed to parse API response as JSON: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)